rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -1705,14 +1709,15 @@ private String retrieveTaskLogsUsingDatabase(String sessionId, String jobId, Str
             throws NotConnectedRestException, UnknownJobException, UnknownTaskException, NotConnectedException,
             PermissionException, PermissionRestException {
         Scheduler scheduler = checkAccess(sessionId, "jobs/" + jobId + "/tasks/" + taskName + "/result/log/all");
-
-        TaskResult taskResult = scheduler.getTaskResult(jobId, taskName);
-
-        if (taskResult != null && taskResult.getOutput() != null) {
-            return taskResult.getOutput().getAllLogs(true);
+        StringBuilder allLogs = new StringBuilder();
```

Comment:
This block of code seems repeated few times. Maybe we can extract a single private method ?

Reply:
it changes each time getStdoutLogs(), getStderrLogs(), getAllLogs()